### PR TITLE
Feat/revamp chain creation flow

### PR DESCRIPTION
**Pull request by vidvidvid**
Implemented a whole bunch of design changes!
### Design: https://www.figma.com/file/SNuBurDIoPjPgkp9LGn6m5/Quests-UI?node-id=559%3A6055
The flow was tested on rinkeby.

#### New explainer screen
Screenshot 2022-08-23 at 19 31 54

#### step 1 - Metadata
Screenshot 2022-08-23 at 19 36 48

#### step 2 - NFTs
- modal opens when switching to 3D or custom NFT tab, warning users that they'll have to pay for it
Screenshot 2022-08-23 at 19 40 00

#### step 3 - Roles
- cannot add an address if it's not a valid address
Screenshot 2022-08-24 at 19 54 38

#### step 4 - Quests
- users can now add multiple quests to the quest chain in a single transaction!!
- if the creator changes their mind, they can go back to 2nd step and take a free NFT instead of 3D or custom which are payable
- user first has to approve tokens
- then they make the transaction
Screenshot 2022-08-23 at 19 42 25

- which is followed by navigating to the newly created quest chain!
Screenshot 2022-08-24 at 19 55 53

### CODE REVIEW

File: components/CreateChain/RolesForm.tsx [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+                  Admin
+                </Text>
+                <Text fontSize={14} fontWeight="bold">
+                  Editor
+                </Text>
+                <Text fontSize={14} fontWeight="bold">
+                  Reviewer
+                </Text>
+                <Text fontSize={14} fontWeight="bold" pr={8}>
+                  Add/remove owners & upgrade to premium chain
```
Comment (dan13ram) [inline components/CreateChain/RolesForm.tsx:150]:
```suggestion
                  Add/remove owners or admins, and upgrade to premium chain
```

File: components/CreateChain/RolesForm.tsx [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+                </Text>
+                <CheckIcon />
+                <CloseIcon color="gray.600" />
+                <CloseIcon color="gray.600" />
+                <CloseIcon color="gray.600" />
+                <Text fontSize={14} fontWeight="bold" pr={8}>
+                  Add/remove admins, editors and reviewers
```
Comment (dan13ram) [inline components/CreateChain/RolesForm.tsx:157]:
```suggestion
                  Add/remove editors or reviewers
```

File: components/CreateChain/RolesForm.tsx [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+  adminAddresses: string[];
+  editorAddresses: string[];
+  reviewerAddresses: string[];
+}
+
+export const RolesForm: React.FC<{
+  onSubmit: (values: RolesFormValues) => void | Promise<void>;
+}> = ({ onSubmit }) => {
+  const initialValues: RolesFormValues = {
+    ownerAddresses: [''],
```
Comment (dan13ram) [inline components/CreateChain/RolesForm.tsx:44]: why are they starting with empty strings ? could be just empty arrays right ?

File: graphql/globalInfo.ts [hunk 4]
```diff
@@ -8,7 +8,12 @@ import { SUPPORTED_NETWORK_INFO } from '@/web3';
 
 export const getFactoryAddress = async (
```
[the remark this answers is not in the document]
Comment (vidvidvid) [reply graphql/globalInfo.ts:9]: true, will do!

**Review by dan13ram — APPROVED** [review]

**Pull request merged**